.github/workflows/go.yml

```diff
@@ -2,14 +2,18 @@ name: Go Build and Release
 
 on:
   push:
-    branches: [ "main" ]
+    branches: ["main"]
+    tags:
+      - "v*"
   pull_request:
-    branches: [ "main" ]
+    branches: ["main"]
 
 jobs:
   build:
     name: Build binaries for Windows, macOS, and Linux
     runs-on: ${{ matrix.os }}
+    permissions:
+      contents: write
```

Comment:
Should provide the workflow with permissions to upload assets to GitHub releases.